reader.go

```diff
@@ -799,12 +802,7 @@ func (r *Reader) partitionWatcher(conn *Conn) func(stop <-chan struct{}) {
 			})
 			return
 		}
-		// It's possible that the list of partitions returned are not in any order
-		// so put them into a map to compare.
-		oParts := make(map[int]struct{})
-		for _, p := range ops {
-			oParts[p.ID] = struct{}{}
-		}
+		oParts := make([]int, len(ops))
```

Comment:
After the latest round of changes, I think we can go even one farther and just keep a count.  What do you think?

Reply:
Yeah, I think that would work. 
Let me get that unit test working and I can see how it looks.